carbon-design-system/carbon-for-ibm-dotcom
# feat(background-media): add "Background media" UI component

### Related Ticket(s)
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6129

DDS Consulting Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6394
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2014

### Description
- This PR adds the "Background media" UI component/pattern that designed to display an `image` or `video` with gradient options. For MVP we're looking to expose `image` for now, and later should be enhanced to support video background.
- This component is dependent of the L-shaped card group - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6129.

### Design & Functional Specs
- https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/wiki/Background-media

![68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3562653039633361363130323930306336643136643939312f31643833633730322d383565342d346165352d383636342d343364326330326633653735](https://user-images.githubusercontent.com/1815714/124028191-b5fb6880-d9c1-11eb-94dd-9195cbdeea12.png)



### Changelog

**New Components/Props**

- `<dds-background-media>` – this new layout component is a variant/extension of the `<dds-image>`.
- `gradient-direction` - a prop/attribute (left-to-right (default) | top-to-bottom)
- `mobile-position` - a prop/attribute (top (default) | bottom)

**Removed**

- N/A

**Note**

- Need to figure out how to treat mobile image position where typically this should be defined within the layout component this `<dds-background-media>` is being used.


**Testing Instructions**
- Visit the `Background media` - `/?path=/story/components-background-media--default`
- Check the "Gradient direction" knobs
- Ensure that the component matches with the design specs for all breakpoints.
